cdap-ui/app/directives/log-viewer-preview/log-viewer-preview.html

```diff
@@ -0,0 +1,215 @@
+            in-view="LogViewerPreview.inViewScrollUpdate($index, $inview, $event)"
+            in-view-options="{ debounce : 500}">
+
+          <td class="time time-stack-trace"
+              ng-if="!entry.stackTrace && entry.log.stackTrace.length > 0 && LogViewerPreview.configOptions.time && !LogViewerPreview.hiddenColumns.time">
+            <span class="indicator-arrow" ng-if="entry.log.stackTrace.length > 0">
+              <i ng-if="($index+1 < LogViewerPreview.displayData.length) && (LogViewerPreview.displayData[$index +1].stackTrace )" class="fa fa-chevron-down"></i>
+              <i ng-if="($index+1 >= LogViewerPreview.displayData.length) || !LogViewerPreview.displayData[$index +1].stackTrace" class="fa fa-chevron-right"></i>
+            </span>
+            <span class="display-time" ng-bind="entry.log.displayTime"></span>
+          </td>
+
+          <td class="time"
+              ng-if="!entry.stackTrace && entry.log.stackTrace.length === 0 && LogViewerPreview.configOptions.time && !LogViewerPreview.hiddenColumns.time">
+            <span class="display-time" ng-bind="entry.log.displayTime"></span>
+          </td>
+
+          <td class="level"
+              ng-if="!entry.stackTrace && LogViewerPreview.configOptions.level && !LogViewerPreview.hiddenColumns.level"
+              ng-bind="entry.log.logLevel"></td>
+
+          <td class="source"
+              ng-if="!entry.stackTrace && LogViewerPreview.configOptions.source && !LogViewerPreview.hiddenColumns.source"
+              uib-tooltip="{{entry.log.threadName}}:{{entry.log.className}}#{{entry.log.lineNumber}}"
+              tooltip-append-to-body="true"
+              tooltip-placement="top"
+              tooltip-popup-delay="500"
+              tooltip-class="source-tooltip">
+              <span
+                ng-bind="(entry.log.threadName)+':'+(entry.log.className)+'#'+(entry.log.lineNumber)"></span>
+          </td>
+          <td class="log-message"
+              ng-if="!entry.stackTrace && LogViewerPreview.configOptions.message && !LogViewerPreview.hiddenColumns.message"
+              uib-tooltip="{{entry.log.message}}"
+              tooltip-append-to-body="true"
+              tooltip-placement="top"
+              tooltip-popup-delay="500"
+              tooltip-class="message-tooltip">
+            <span ng-bind-html="LogViewerPreview.highlight(entry.log.message)"></span>
+          </td>
+          <td class="stack-trace" ng-if="entry.stackTrace" colspan="4">
+            <pre ng-bind="entry.log.stackTrace"></pre>
+          </td>
+        </tr>
+
+        <tr class="message-holder">
+          <td colspan="4">
+            <div class="well well-sm text-center"
+                 ng-if="!LogViewerPreview.errorRetrievingLogs && !LogViewerPreview.loading && !LogViewerPreview.displayData.length && LogViewerPreview.statusType !== 0">
+              <h4>
+                There are no logs yet to display.
```

Comment:
I feel like this message is actually ambiguous @nitinmotgi. It could be that the program has completed run and there is no logs.